src/openforms/prefill/attributes_generator.py

```diff
@@ -0,0 +1,130 @@
+    * capitalize each part for the human-readable label
+    """
+
+    parts: list[str]
+
+    @property
+    def variable_name(self) -> str:
+        return "_".join(part.lower() for part in self.parts).lstrip("_")
+
+    @property
+    def value(self) -> str:
+        return ".".join(self.parts)
+
+    @property
+    def label(self) -> str:
+        return " > ".join(part.capitalize() for part in self.parts)
+
+
+@dataclass
+class OpenApi3AttributesGenerator:
+    """Generate the attributes from OAS specification
+
+    TODO features present in the generate_prefill_from_spec command
+
+    * Specify a service and then generate the attributes from the OAS of this service
+    * Specify a path and then only generate the attributes for this path
+    * If no path/schema are specified, print all paths and schemas in the OAS
+    * Option to add '_embedded.' in paths for $refs
+
+    Additional features that could be good
+
+    * Keep track of the titles in addition to the properties names in the schema
+    """
+
+    url: str = ""
+    """
+    The URL of the OAS specification
+    """
+    schema: str = ""
+    """
+    The name of the schema in the specification from which to generate the attributes
+    """
+
+    def generate_attributes(self) -> str:
+        oas_specification = self._load_specification()
+        attributes = self._get_schema_attributes(oas_specification, self.schema)
+        return render_from_string(
+            FILE_TEMPLATE,
+            context={
+                "url": self.url,
+                "schema": self.schema,
+                "attributes": attributes,
+            },
+            disable_autoescape=True,
+        )
+
+    def _load_specification(self) -> Specification:
+        if self.url:
+            # The content types supported in the openapi3-parser are these:
+            # https://github.com/manchenkoff/openapi3-parser/blob/master/src/openapi_parser/enumeration.py#L97
+            # However, in the haal centraal OAS we find content types such as "application/hal+json". So with
+            # strict_enum=False, the parser just checks that the content type is a string
+            return parse(uri=self.url, strict_enum=False)
+
+        raise AttributeGeneratorException(
+            "No URL to the OAS specification was provided."
+        )
+
+    def _get_schema_attributes(
+        self, oas_specification: Specification, schema_name: str
+    ) -> Iterator[Attribute]:
+        if (schema := oas_specification.schemas.get(schema_name)) is None:
+            raise AttributeGeneratorException(
+                f'The given schema name "{schema_name}" could not be found in the specifications'
+            )
+
+        assert isinstance(
+            schema, Object
+        ), f"Expected an object schema, got {type(schema)}."
+        for schema_property in schema.properties:
+            yield from self._get_property_attributes(schema_property)
```

Comment:
looking at this method and also `_get_property_attributes`, none of them seem to actually be using `self`, so I'm inclined to make them `@staticmethod`, but that breaks the `self._get_property_attributes` invocation.

Shouldn't we just use plain functions instead of methods then?